### PR TITLE
[Metrics] Remove public metrics port.

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This PR:
1. Removes the public metrics server (e.g., on port `9102`). This is unused.
2. Removes the path `/-/healthy` on the metrics port. This is also unused. The rest of the paths are TBD.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Unit tests.

## Related PRs

None.

### CODE REVIEW

File: aptos-node/src/lib.rs [hunk 1]
```diff
@@ -408,11 +408,6 @@ async fn periodic_telemetry_dump(node_config: NodeConfig, db: DbReaderWriter) {
                 // Build the params from internal prometheus metrics
                 let mut metrics_params: HashMap<String, String> = HashMap::new();
 
-                let met = get_public_json_metrics();
```
[the remark this answers is not in the document]
Comment (rustielin) [reply aptos-node/src/lib.rs:411]: Yea it's fine. We can collect these metrics through other ways. I need to give telemetry a better API too.

[the remark this answers is not in the document]
Comment (rustielin) [reply aptos-node/src/lib.rs:411]: cc @pr-aptos we'll lose the system cpu/memory metrics until I reintroduce them in a separate PR , along with all the new metrics from our schema

[the remark this answers is not in the document]
Comment (JoshLind) [reply aptos-node/src/lib.rs:411]: Thanks @rustielin! We can remove it in this PR and I'll follow up 😄

**Comment by JoshLind** [comment]
Thanks all!

/land

**Pull request closed without merging**